client/src/components/BlockEditor/BlockEditorEditing.tsx

```diff
@@ -195,6 +196,26 @@ export class BlockEditorEditingPresentational extends React.Component<
     }
   };
 
+  private handleSquareBracketExport() {
+    // check to see whether there are a balanced number of square brackets
+    // if there are, everything within the outermost brackets gets exported
+    const didConvertBrackets = slateChangeMutations.scanBlockAndConvertOuterSquareBrackets({
+      change: this.props.value.change(),
+      updateBlock: this.props.updateBlock,
+      exportSelection: this.props.exportSelection,
+      blockId: this.props.block.id,
+    });
```

Comment:
I'm not a big fan of the pattern of mutations return true or false depending on if they ran or not; it's just not obvious to me that ``scanBlockAndConvertOuterSquareBrackets`` would either run or not run, and would return a boolean if it did or not. I generally expect mutations to be silent.

Not sure what the best way to deal with this is. Perhaps if it returned a value like, {wasMutationPerformed: true} instead.

Reply:
I'll go with returning `{wasMutationPerformed: true}`. I don't think it's totally out there to return something to indicate whether a mutation occurred. For example, in Sequelize, the `destroy` method returns an integer indicating the number of destroyed rows. That said, I'd be happy if we could think of a more natural way to do this.

Reply:
This works. I'll kind of keep it in mind, but don't expect to reach an epiphany in the next few days or so.